src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -221,6 +221,14 @@ def validate_spot_max_price(namespace):
                 "default price to be up-to on-demand")
 
 
+def validate_message_of_the_day(namespace):
+    """Validates message of the pool can only be used on Linux."""
+    if namespace.message_of_the_day is not None and namespace.message_of_the_day != "":
+        if hasattr(namespace, 'os_type'):
+            if namespace.os_type is not None and namespace.os_type != "Linux":
+                raise CLIError('--message-of-the-day can only be set for linux nodepools')
+
+
```

Comment:
I suppose this is not a good place to validate multiple options. Especially for command lie `aks create`, there's no such option (`--os-type`).

Reply:
yeah, I tried to approximate with `hasattr`, which works but is not very idiomatic/pythonic. 

would you prefer I revert this?

Reply:
suppose I could split the validation func and have it different for create / nodepool add

Reply:
ah actually, I didn't realize you can't do windows nodepools at all on az aks create. so validation isn't required there.

Reply:
At first I thought we should do some validation against windows profile in the getter function of this option, but then I realized that we can still use this feature on linux node pools.

Never mind, I think adding verification to this option for `nodepool add` will do. But we may need to leave a note for `aks create`.